src/util/getIdsFromJobState.ts

```diff
@@ -0,0 +1,7 @@
+export async function getIdsFromJobState(_type, jobState) {
```

Comment:
Related to other comments, I think this may be a signal that other code needs to change to just iterate the entities, without having to collect all their ids and iterate those.